tests/installation/addon_products_via_SCC_yast2.pm

```diff
@@ -15,29 +15,51 @@ use base qw(y2logsstep y2x11test);
 use strict;
 use testapi;
 use registration 'fill_in_registration_data';
+use version_utils 'sle_version_at_least';
 
-sub run {
-    my ($self) = @_;
-    x11_start_program('xterm');
+sub prepare_reg_sle12 {
     # add every used addon to regurl for proxy SCC
     if (get_var('SCC_ADDONS')) {
         my @addon_proxy = ("url: http://server-" . get_var('BUILD_SLE'));
         for my $addon (split(/,/, get_var('SCC_ADDONS', ''))) {
             my $uc_addon = uc $addon;    # change to uppercase to match variable
             push(@addon_proxy, "\b.$addon-" . get_var("BUILD_$uc_addon"));
         }
-        script_sudo "echo \"@addon_proxy.proxy.scc.suse.de\" > /etc/SUSEConnect", 0;
+        # Define proxy SCC
+        assert_script_run "echo \"@addon_proxy.proxy.scc.suse.de\" > /etc/SUSEConnect";
     }
-    # Disable screensaver
-    type_string "gsettings set org.gnome.desktop.session idle-delay 0\n";
-    send_key "ctrl-d";
-    $self->launch_yast2_module_x11('scc', target_match => [qw(scc-registration packagekit-warning)]);
+
+}
+
+sub prepare_reg_sle15 {
+    # Remove registration from the system
+    assert_script_run 'SUSEConnect --clean';
+    # Define proxy SCC
+    assert_script_run 'echo "url: ' . get_var('SCC_URL') . '" > /etc/SUSEConnect';
+}
+
+sub test_setup {
+    select_console 'root-console';
+    if (sle_version_at_least '15') {
+        prepare_reg_sle15;
```

Comment:
so this will still be called for SLE16 but calls a function "reg_sle15"? Somehow does not fit nicely. Better call the functions with a name reflecting what they really do